kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/SchemaUtils.java

```diff
@@ -436,6 +440,123 @@ private static Map<Long, StructField> fieldsById(StructType schema) {
     return columnIdToField;
   }
 
+  private static void validateNestedIds(
```

Comment:
Can you add docs for this fx?

Checks that all fields that should have nested ids have them? And that they are consistent?